twitchio/ext/eventsub/models.py

```diff
@@ -238,6 +238,77 @@ def __init__(self, client: EventSubClient, data: dict):
         self.is_gift: bool = data["is_gift"]
 
 
+class ChannelSubscriptionGiftData(EventData):
+    """
+    A Subscription Gift event
+    Explicitly, the act of giving another user a Subscription.
+    Receiving a gift-subscription uses ChannelSubscribeData above, with is_gift==True
+
+    Attributes
+    -----------
+    is_anonymous: :class:`bool`
+        Whether the gift sub was anonymous
+    user: Optional[:class:`twitchio.PartialUser`]
+        The user that gifted subs. Will be ``None`` if ``is_anonymous`` is ``True``
+    broadcaster: :class:`twitchio.PartialUser`
+        The channel that was subscribed to
+    tier: :class:`int`
+        The tier of the subscription
+    total: :class:`int`
+        The total number of subs gifted by a user at once
+    cumulative: Optional[:class:`int`]
+        The total number of subs gifted by a user overall. Will be ``None`` if ``is_anonymous`` is ``True``
+    """
+
+    __slots__ = "is_anonymous", "user", "broadcaster", "tier", "total", "cumulative"
+
+    def __init__(self, client: EventSubClient, data: dict):
+        self.is_anonymous: bool = data["is_anonymous"]
+        self.user: Optional[PartialUser] = _transform_user(client, data, "user") if not self.is_anonymous else None
+        self.broadcaster = _transform_user(client, data, "broadcaster_user")
+        self.tier = int(data["tier"])
+        self.total = int(data["total"])
+        self.cumulative = int(data["cumulative_total"]) if not self.is_anonymous else None
```

Comment:
This would be better named as cumulative_total to match the response and avoid any confusion.
This is a convention that is followed in majority of the lib.